go/exocom/src/exocom/exocom.go

```diff
@@ -140,14 +140,12 @@ func (e *ExoCom) send(message structs.Message) error {
 		printLogError(e.logger.Warning(fmt.Sprintf("Warning: No receivers for message '%s' registered", publicMessageName)))
 		return nil
 	}
-	if message.ResponseTo != "" {
-		originalTimestamp, ok := e.messageCache.Get(message.ID)
-		if ok {
-			message.ResponseTime = message.Timestamp.Sub(originalTimestamp)
-		}
-	} else {
-		e.messageCache.Set(message.ID, message.Timestamp)
+
+	originalTimestamp, ok := e.messageCache.Get(message.ID)
```

Comment:
We need to update the message cache to be a mapping from message.ActivityID to timestamp. The response time will be based on the last message with the same activityID.